docs/contribute/addnode.qmd

```diff
@@ -0,0 +1,170 @@
+---
+title: "How to add a new node type"
+---
+
+Several parts of the code have to be made aware of the new node type. In the rest of this page we shall call our new node type `NewNodeType`.
+
+# The Julia core
+
+## Parameters
+
+The parameters object (defined in `solve.jl`) passed to the ODE solver must be made aware of the new node type. Therefore define a struct in `solve.jl` which holds the data for each node of the new node type:
+
+```julia
+struct NewNodeType
+    node_id::Vector{Int}
+    # Other fields
+end
+```
+
+This struct should at least contain a vector of all node IDs that are of the new type in a given model. Now you can:
```

Comment:
Perhaps it would be good to clarify that the fields in this struct are whatever is needed data wise during the simulation, and that this doesn't have to correspond 1:1 with the input tables.